makeflow/src/makeflow_status.c

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright (C) 2016- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+
+/* 
+ * File:   makeflow_status.c
+ * Author: Kyle D. Sweeney
+ *
+ * Created on July 21, 2016
+ */
+
+
+#include "timestamp.h"
+#include "jx_table.h"
+#include "jx_print.h"
+#include "jx_parse.h"
+#include "stringtools.h"
+#include "list.h"
+#include "catalog_query.h"
+#include "json.h"
+#include "json_aux.h"
+#include "getopt_aux.h"
+#include "cctools.h"
+#include "xxmalloc.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+static void show_help(const char *cmd)
+{
+	fprintf(stdout, "makeflow_status [options]\n");
+	fprintf(stdout, "where options are:\n");
+	fprintf(stdout, " %-30s Find all from this user.\n", "-u,--username=<user>");
+        fprintf(stdout, " %-30s Find this project name.\n", "-N, --project=<proj>");
+        fprintf(stdout, " %-30s Use this Server address.\n", "-s, --server=<server>");
+        fprintf(stdout, " %-30s Use this Port on the server.\n", "-p, --port=<port>");
+	fprintf(stdout, " %-30s Timeout.\n", "-t,--timeout=<time>");
+	fprintf(stdout, " %-30s This message.\n", "-h,--help");
+}
+
+int compare_entries(struct jx **a, struct jx **b)
+{
+	int result;
+	const char *x, *y;
+
+        // by project
+	x = jx_lookup_string(*a, "project");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "project");
+	if(!y)
+		y = "unknown";
+
+	result = strcasecmp(x, y);
+	if(result != 0)
+		return result;
+        
+        //by owner
+        x = jx_lookup_string(*a, "owner");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "owner");
+	if(!y)
+		y = "unknown";
+
+	result = strcasecmp(x, y);
+	if(result != 0)
+		return result;
+
+        //failsafe by servername
+	x = jx_lookup_string(*a, "name");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "name");
+	if(!y)
+		y = "unknown";
+
+	return strcasecmp(x, y);
+}
+
+static struct jx *table[10000];
+
+int first_comma(char* in){
+    int i=0;
+    int max = strlen(in);
+    for(i=0; i<max; ++i){
+        if(in[i] == ',')
+            return i;
+    }
+    return -1;
+}
+char* add_port(char* in, int port){
```

Comment:
I see you've noticed the issues with a single port option applied to multiple hosts. It's best to avoid parsing the host list yourself, there's logic elsewhere that handles it, so you can just treat it as an opaque string.